Arelle/Arelle
# Make FR-NL-1.01 validation encoding independent

#### Description of change

- Previous FR-NL-1.01 validation failed when passing hardcoded encoding or leaving Arelle to determine encoding (Arelle's file loading process will trim the BOM character under certain conditions).
	- Corrected by passing `binary=True` to operate independent of any encoding. Check against BOM constants from codecs to match against any feasible BOM byte sequence. 
- FR-NL-1.01 and FR-NL-1.05 had mismatched test case filenames.
	- Renamed accordingly
- The silent failure of missing references in test case indexes has been problematic as of late.
	- Added assertion that will fail if `IOerror` is encountered when parsing test cases index 

#### Steps to Test

Checkout locally and invalidate a test case path in the NT18 index file. Run NT18 conformance suite, should fail with AssertionError.

**review**:
@Arelle/arelle
